velox/functions/prestosql/RandN.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "folly/Random.h"
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions {
+
+template <typename T>
+struct RandNFunction {
+  static constexpr bool is_deterministic = false;
+
+  template <typename TOutput, typename TInput = TOutput>
+  FOLLY_ALWAYS_INLINE void call(TOutput& out, const TInput& input) {
+    if constexpr (std::is_same_v<TInput, int64_t>) {
```

Comment:
Looks like we need to add some checks for 'input':

```
presto:di> select random(-10);

Query 20230404_180051_72062_5mix7 failed: bound must be positive

presto:di> select random(0);

Query 20230404_180111_72086_5mix7 failed: bound must be positive
```

Reply:
Added `VELOX_USER_CHECK_GT`